src/Alba/ScenarioAssertionException.cs

```diff
@@ -65,11 +66,17 @@ public override string Message
         }
 
         /// <summary>
-        /// A textual representation of the HTTP response body for diagnostic purposes
+        /// A textual representation of the HTTP response body for diagnostic purposes. This property will be null unless <see cref="ReadBody"/> is called.
         /// </summary>
         public string? Body { get; set; }
 
-        internal string ReadBody(HttpContext context)
+        /// <summary>
+        /// Reads the response body and returns it as a string
+        /// </summary>
+        /// <param name="context">The context of the response</param>
+        /// <returns>A string with the content of the body</returns>
+        [MemberNotNull(nameof(Body))]
+        public string ReadBody(HttpContext context)
```

Comment:
It's kind of annoying that I have to call `ReadBody` to assign the `Body` property and its not very discoverable to end users that this is required and that there is a relationship between the two.

The ReadBody method itself is useful since it will rewind the stream back to the beginning, which could affect other assertions helpers. Perhaps ReadBody should actually be an extension method on HttpContext instead and have a similar to the helper methods in `System.Net.Http.Json`, e.g. `ReadResponseBodyAsText`.